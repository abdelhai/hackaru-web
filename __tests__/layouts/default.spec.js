import { Store } from 'vuex-mock-store';
import Factory from '@/__tests__/__setups__/factory';
import MockDate from 'mockdate';
import Default from '@/layouts/default';

describe('Default', () => {
  let factory;
  let wrapper;

  MockDate.set('2019-01-31T01:23:45');

  const $store = new Store({
    getters: {
      'activities/getWorkingActivities': () => []
    }
  });

  beforeEach(() => {
    $store.reset();

    factory = new Factory(Default, {
      stubs: ['nuxt'],
      mocks: { $store }
    });
  });

  it('render correctly', () => {
    expect(factory.shallow().element).toMatchSnapshot();
  });

  it('dispatch projects/getProjects', () => {
    factory.shallow();
    expect($store.dispatch).toHaveBeenCalledWith('projects/getProjects');
  });

  describe('when call updateDuration and has no working activities', () => {
    beforeEach(() => {
      $store.getters['activities/getWorkingActivities'] = [];
      wrapper = factory.shallow();
      wrapper.vm.updateDuration();
    });

    it('does not have titleTemplate', () => {
      expect(wrapper.vm.$meta().inject().titleTemplate).toBeUndefined();
    });
  });

  describe('when call updateDuration and has working activities', () => {
    beforeEach(() => {
      $store.getters['activities/getWorkingActivities'] = [
        {
          id: 1,
          description: 'Review',
          startedAt: '2019-01-01T01:23:45'
        }
      ];
      wrapper = factory.shallow();
      wrapper.vm.updateDuration();
    });

    it('has titleTemplate correctly', () => {
      expect(wrapper.vm.$meta().inject().titleTemplate).toBe('720:00:00ãƒ»%s');
    });
  });
});
