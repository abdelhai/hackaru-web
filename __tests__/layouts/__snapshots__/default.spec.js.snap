// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default render correctly 1`] = `
<div
  class="wrapper"
  ontouchstart=""
>
  <nav-modal-stub
    height="450"
    initialcomponent="[object Object]"
    keepalives="ActivityEditor"
    name="activity"
  />
   
  <side-bar-stub
    showmenu="true"
  />
   
  <login-guard-stub
    class="content"
  >
    <nuxt-stub />
     
    <pwa-balloon-stub />
  </login-guard-stub>
</div>
`;
